server/middleware/rest.js

```diff
@@ -23,9 +23,15 @@ module.exports = rest;
  */
 
 function rest() {
+  var router; // Cache
+
   return function restApiHandler(req, res, next) {
     var app = req.app;
-    var restHandler = app.handler('rest');
+    var restHandler;
+
+    if (!router) {
+      restHandler = app.handler('rest');
+    }
```

Comment:
This is a breaking change. Note that `app.handler('rest')` is already caching the handler fn. However, when a new model is added, this cache is cleared, so that the next call of `app.handler('rest')` builds a new handler fn that includes the new model too.